src/JoyOI.Monitor/Controllers/ManagementService/StateMachineController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+
+using JoyOI.Monitor.Models;
+using System.Drawing;
+
+namespace JoyOI.Monitor.Controllers.ManagementService
+{
+    [Route("/ManagementService/StateMachine")]
+    public class StateMachineController : ChartController
+    {
+        const string MGMTSVC = "mgmtsvc";
+
+        [HttpGet("Created")]
+        public async Task<IActionResult> Created(int start, int end, int interval, int timezoneoffset)
+        {
+            if (start == 0 || end == 0 || interval == 0) {
+                Response.StatusCode = 400;
+                return Json(null);
+            }
+            var scaling = new ChartScaling(start, end, interval);
+            return Json(await GetChartData(
+                MGMTSVC,
+                "SELECT " +
+                "FLOOR(UNIX_TIMESTAMP(StartTime) / @interval) * @interval as t, " +
+                "Count(Id) as c " +
+                "FROM joyoi_mgmtsvc.statemachineinstances " +
+                "GROUP BY t " +
+                "HAVING t >= @start AND t <= @end " +
+                "ORDER BY t DESC " +
+                "LIMIT 0, @points",
+                scaling,
+              (rows) =>
+              {
+                  var rows_tuple =
+                    rows.Select(d => Tuple.Create(
```

Comment:
```c#
d => (d.GetInt64("t"), d.GetDouble("c"))
```

Reply:
`d` is a dictionary here

Reply:
So, it should be d => (Convert.ToInt64(d["t"]), Convert.ToDouble(d["c"]))